Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -2556,10 +2556,18 @@ def read(self, n):  # type: ignore[override]
     def readinto(self, b):
         sz = len(b)
         data = self.responseReader.read(sz)
-        for idx, ch in enumerate(data):
+
+        # Remove non utf-8 characters to avoid decode errors in JSONResultsReader
+        cleaned_data = data.decode("utf-8", errors="ignore").encode("utf-8")
```

Comment:
Can you please add a link to the splunk recommendation about how to handle this use case?